### ISSUE TITLE
[Feature request] Possibility to add global variables to the list of known variables in plugin configuration

### ISSUE THREAD

**Issue by g0ga** [Camelcade/Perl5-IDEA#2157]
# [Feature request] Possibility to add global variables to the list of known variables in plugin configuration

Many projects have variables imported using boiler plate used everywhere, which can happen through Import::Into or plain name space modification. So there's no way to figure out if such a variable has a declaration or not, which lead to a situation that syntax check says there are undeclared variables, which is not true. So i'd suggest to add a list of variables to exclude from the inspection, same way it is done for instance variables in plugin configuration (since that inspection is one of the most useful)

**Comment by hurricup** [comment]
Instance variables are introduced to distinct methods from just functions. They are not just silently declared.
I'm always trying to solve problems in proper way, not just do some shortcuts.
Do I get this right and you want support for `Import::Into` module?

**Comment by g0ga** [comment]
As i said, there's a million of ways for importing something into another package's name space. Our project has $log variable imported everywhere like this from logger class
```sub import {
    my $class = shift;
    return if ! $_[0] =~ /^\$/

    $_[0] =~ s/^\$//;
    my $package = (caller)[0];
    no strict 'refs';
    *{"$package\::$_[0]"} = \($class->new());
}
```
and then when using it:
use Something::Logger qw($log);
will get you that variable imported. So i'd just want a way to have those variable be marked as declared.

**Comment by hurricup** [comment]
Yes, I'm aware that this is perl and it does things in million different ways. 
What do you suggest to do with methods of your logger? We can whitelist `$log` but what about methods?

**Comment by g0ga** [comment]
Hell with those methods. Unresolved sub inspection is kinda useless anyways, since a ton of modules uses autoload or generated methods. While unresolved variable is usually something bad, probably a leftover or a mistake, which will be pointed at on review. If you care about methods you can do it like a whitelist map $var -> class that will cover all possible use cases.

**Comment by todd-richmond** [comment]
would also be nice to allow disable some of the 'known' globals - specifically $a and $b as we use my $a for arrayrefs in some places and $b as booleans often

**Comment by hurricup** [comment]
@todd-richmond what do you mean to `disable` them?

**Comment by todd-richmond** [comment]
I just mean be able to remove those vars from the predefined list so they don't show up as inspection warnings. Any other solution to prevent warnings for $a, $b would be fine too
